pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentSubscription.java

```diff
@@ -307,6 +307,8 @@ public synchronized void removeConsumer(Consumer consumer, boolean isResetCursor
         msgOutFromRemovedConsumer.add(stats.msgOutCounter);
 
         if (dispatcher != null && dispatcher.getConsumers().isEmpty()) {
+            // clear delay message avoid duplicate.
+            dispatcher.clearDelayedMessages();
```

Comment:
There are other places will call `cursor.rewind();` in the PersistentDispatcherMultipleConsumers. The cursor rewind will cause to add the duplicated delayed message index, and reset the cursor will also introduce this problem, could you please check them?